pkg/addon/manager/addon.go

```diff
@@ -183,6 +185,25 @@ func newServiceAccount(namespace string) *corev1.ServiceAccount {
 	}
 }
 
+func newAddonAgentClusterRole(namespace string) *rbacv1.ClusterRole {
```

Comment:
i wonder if we can move the tokenreview roles to a namespaced `Role` instead of `ClusterRole`, which is also where we placed the tokenrequest roles (the `newAddonAgentRole` func below).

Reply:
```
r.SpokeNativeClient.AuthenticationV1().TokenReviews().Create()
```
is a cluster scoped request, and I didn't find it in serviceaccount APIs where the tokenrequest is located  😢